Unused variables breaking build

Hello.

Using the default compiler (gcc (GCC) 5.1.0) on an up-to-date Arch Linux install, a plain `make` of 6ace3b655cf13dbd1e23b2535c51abf3305fea63 with no parameters results in:

```
compiling build/linux-x86_64/machine.o
In file included from src/avian/processor.h:18:0,
                 from src/avian/machine.h:21,
                 from src/avian/jnienv.h:14,
                 from src/machine.cpp:11:
src/avian/bootimage.h:64:17: error: ‘vm::TargetBootFlatConstant’ defined but not used [-Werror=unused-variable]
 #define NAME(x) Target##x
                 ^
src/bootimage-template.cpp:6:16: note: in expansion of macro ‘NAME’
 const unsigned NAME(BootFlatConstant) = 1 << NAME(BootShift);
                ^
src/avian/bootimage.h:64:17: error: ‘vm::TargetBootHeapOffset’ defined but not used [-Werror=unused-variable]
 #define NAME(x) Target##x
                 ^
src/bootimage-template.cpp:7:16: note: in expansion of macro ‘NAME’
 const unsigned NAME(BootHeapOffset) = 1 << (NAME(BootShift) + 1);
                ^
In file included from src/avian/processor.h:18:0,
                 from src/avian/machine.h:21,
                 from src/avian/jnienv.h:14,
                 from src/machine.cpp:11:
src/bootimage-template.cpp:6:21: error: ‘vm::BootFlatConstant’ defined but not used [-Werror=unused-variable]
 const unsigned NAME(BootFlatConstant) = 1 << NAME(BootShift);
                     ^
src/avian/bootimage.h:70:17: note: in definition of macro ‘NAME’
 #define NAME(x) x
                 ^
src/bootimage-template.cpp:7:21: error: ‘vm::BootHeapOffset’ defined but not used [-Werror=unused-variable]
 const unsigned NAME(BootHeapOffset) = 1 << (NAME(BootShift) + 1);
                     ^
src/avian/bootimage.h:70:17: note: in definition of macro ‘NAME’
 #define NAME(x) x
                 ^
cc1plus: all warnings being treated as errors
makefile:1804: recipe for target 'build/linux-x86_64/machine.o' failed
make: *** [build/linux-x86_64/machine.o] Error 1
```

I'm guessing this version of gcc produces more warnings than the developer(s) compilers, and the `-Werror` flag is causing issues here.

--- a/src/bootimage-template.cpp
+++ b/src/bootimage-template.cpp
@@ -1,10 +1,7 @@
 const unsigned NAME(BootMask) = (~static_cast<unsigned>(0))
                                 / NAME(BytesPerWord);
 
-const unsigned NAME(BootShift) = 32 - avian::util::log(NAME(BytesPerWord));
-
-const unsigned NAME(BootFlatConstant) = 1 << NAME(BootShift);
-const unsigned NAME(BootHeapOffset) = 1 << (NAME(BootShift) + 1);
+const unsigned NAME(BootShift) UNUSED = 32 - avian::util::log(NAME(BytesPerWord));
 
 inline unsigned LABEL(codeMapSize)(unsigned codeSize)
 {

--- a/src/codegen/compiler/event.cpp
+++ b/src/codegen/compiler/event.cpp
@@ -1111,10 +1111,10 @@ class CombineEvent : public Event {
                 "%p %p %d : %p %p %d\n",
                 secondValue,
                 secondValue->source,
-                secondValue->source->type(c),
+                static_cast<int>(secondValue->source->type(c)),
                 secondValue->nextWord,
                 secondValue->nextWord->source,
-                secondValue->nextWord->source->type(c));
+                static_cast<int>(secondValue->nextWord->source->type(c)));
       }
     }
 

